parser.go

```diff
@@ -1225,8 +1225,20 @@ func (p *Parser) parseIdent(desc string) (*Ident, error) {
 
 func (p *Parser) parseType() (_ *Type, err error) {
 	var typ Type
-	if typ.Name, err = p.parseIdent("type name"); err != nil {
-		return &typ, err
+	for p.peek() == IDENT {
+		typeName, err := p.parseIdent("type name")
+		if err != nil {
+			return &typ, err
+		}
+		if typ.Name == nil {
+			typ.Name = typeName
+		} else {
+			typ.Name.Name += " " + typeName.Name
+		}
+	}
+
+	if typ.Name == nil {
```

Comment:
The only way this will not be nil is if `p.peek() != IDENT`, right? If so should you just roll this into the if block above?

Reply:
The block above is a for loop, not an if statement